File: javapatterns/src/main/java/org/iut/nantes/facade/cours/Retroprojecteur.java
package org.iut.nantes.facade.cours;

public class Retroprojecteur {
    public void brancherVideoprojecteur(Multiprises multiprises) {
        System.out.println("Retroprojecteur branché.");
    }

    public void débrancherVideoprojecteur(Multiprises multiprises) {
        System.out.println("Retroprojecteur débranché.");
    }

    public void allumerVidéoprojecteur() {
        System.out.println("Retroprojecteur allumé.");
    }

    public void éteindreVidéoprojecteur() {
        System.out.println("Retroprojecteur éteint.");
    }
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/facade/cours/Retroprojecteur.kt
package org.iut.nantes.facade.cours

class Retroprojecteur {
    fun brancherVideoprojecteur(multiprises: Multiprises?) {
        println("Retroprojecteur branché.")
    }

    fun débrancherVideoprojecteur(multiprises: Multiprises?) {
        println("Retroprojecteur débranché.")
    }

    fun allumerVidéoprojecteur() {
        println("Retroprojecteur allumé.")
    }

    fun éteindreVidéoprojecteur() {
        println("Retroprojecteur éteint.")
    }
}